### ISSUE TITLE
Media field does not allow to select files that are not images (PDF, ...)

### ISSUE THREAD

**Issue by garstud** [joomla/joomla-cms#19954]
### Steps to reproduce the issue
- verify that you Media Manager accept PDF files (it is ok by default)
- Create an article Media Custom Field 

Use it in an article :
- click on the Media Custom Field
- upload a PDF file
- try to select the uploaded PDF file

### Expected result
You can select the PDF file uploaded through the Media Custom Field because it is accepted and store by the Media Manager.

### Actual result
You succeed to upload the PDF file through the Media Custom Field (you can verify it by the Media Manager)
But you can not see it nor select it through the Media Custom Field.

### System information (as much as possible)
dbversion: 5.7.14
dbcollation: utf8_general_ci
dbconnectioncollation: utf8mb4_general_ci
phpversion: 7.0.10
server: Apache/2.4.23 (Win64) PHP/7.0.10
sapi_name: apache2handler
version: Joomla! 3.8.5 Stable [ Amani ] 6-February-2018 15:00 GMT
platform: Joomla Platform 13.1.0 Stable [ Curiosity ] 24-Apr-2013 00:00 GMT
useragent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/64.0.3282.186 Safari/537.36

**Comment by Quy** [comment]
Duplicate #10871. 

Short answer: Media custom field is for images only.

**Comment by ghost** [comment]
closed as duplicate Report.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/19954">issues.joomla.org/tracker/joomla-cms/19954</a>.</sub>

**Comment by Robdebert** [comment]
Guys, there are so many Issues regarding the possibiliy to select other files types in a modal-window like the media-type allows it, but all the ones that i came accros and read are closed because "duplicate of xy" or "use another field-type".

As you can see in this conversation https://github.com/joomla/joomla-cms/issues/10871 there IS a need to use a modal-window for not only uploading files but also select them after upload. The file-type-field is not a proper solution for this (because you have to upload a file again which might already be on the server - please don't suggest to use 2 fields: file and filelist :-D ).

So: Where is the issue that we can talk about this feature? Please add a link to all the closed ones.

**Comment by ghost** [comment]
@Robdebert thanks for your Comment, but Comments on closed Issues didn't get much Notice. So i suggest to open an new one.

> Please add a link to all the closed ones.

i guess you know but to be sure: open https://github.com/joomla/joomla-cms/issues?q=is%3Aissue+is%3Aclosed

**Comment by Robdebert** [comment]
Thanks for your answer.

But can we not just re-open this issue? Because there is already a lot of information here. Opening a new one will only create the next issue to the same topic, that already exists (in this and other issues, that where created).

**Comment by ghost** [comment]
Reopened as stated above.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/19954">issues.joomla.org/tracker/joomla-cms/19954</a>.</sub>

**Comment by Quy** [comment]
PR #19967 is closed, but addresses this issue.

**Comment by martyharty** [comment]
In Joomla 4 uploading pdf is enabled by default, but not rendered on the frontend. 

Changing plugins/fields/media/tmpl/media.php from:
<code>
...
if (!$path)
{
	continue;
}
$buffer .= sprintf('<img src="%s"%s>',
	htmlentities($path, ENT_COMPAT, 'UTF-8', true),
	$class
);
...
</code>
To:
<code>
...
if (!$path)
{
	continue;
}
if(substr($path, -4) == '.pdf')
{
	$parts = explode('/', $path);
	$buffer = '<a href="'. $path .'" target="_blank" rel="noopener">'. $parts[1] .'</a>';
} else {	
$buffer .= sprintf('<img src="%s"%s>',
	htmlentities($path, ENT_COMPAT, 'UTF-8', true),
	$class
	);	
}
...
</code>
Did the trick for me: https://gridx.joomla.com/

Both are Media fields and Download renders without preview (No) in frontend editor.

The same is done in Joomla 3 without automatic display, custom fields and rawvalue.

In my opinion (pre)setting the images/files/docs folders should work in most cases.

Thanks
<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/19954">issues.joomla.org/tracker/joomla-cms/19954</a>.</sub>

**Comment by martyharty** [comment]
With href:
<pre>
if (!$path)
	{
		continue;
	}
	if(substr($path, -4) == '.pdf')
	{
		$parts = explode('/', $path);
		$buffer = '<&#8205;a href="'. $path .'" target="_blank" rel="noopener">'. $parts[1] .'</a>';
	} else {	
	$buffer .= sprintf('<img src="%s"%s>',
		htmlentities($path, ENT_COMPAT, 'UTF-8', true),
		$class
		);	
	}
</pre><hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/19954">issues.joomla.org/tracker/joomla-cms/19954</a>.</sub>

**Comment by martyharty** [comment]
Attached improved code to always get the end-path from different files folders. https://gridx.joomla.com/
Changing plugins/fields/media/tmpl/media.php to:
```
foreach ($value as $path)
{
	if (!$path)
	{
		continue;
	}
	if(substr($path, -4) == '.pdf')
	{
		$file = explode('/', $path);
		$buffer = '<a href="'. $path .'" target="_blank" rel="noopener">'. end($file) .'</a>';
	} else {
	$buffer .= sprintf('<img src="%s"%s>',
		htmlentities($path, ENT_COMPAT, 'UTF-8', true),
		$class
		);
	}
}
```

**Comment by ghost** [comment]
@martyharty can you do a Pull Request - https://extensions.joomla.org/blog/item/how-to-contribute-to-joomla-through-github/ ?

**Comment by hardik-codes** [comment]
@franz-wohlkoenig can I do this if @martyharty doesn't have any problem

**Comment by ghost** [comment]
@hardik-codes sure.

**Comment by martyharty** [comment]
Please go ahead. I noticed setting a files path on Media Field is not accessible from frontend. Check with user(name) and pass(word) on https://gridx.joomla.com/ (default images path is no problem).

Regarding Joomla 4 it seems the System - Privacy Consent plugin is not redirecting user on frontend login. 
Also saving Content Fields on backend: An error has occurred. 0 Class 'FieldsHelper' not found.

One more thing. The Joomla 4 alpha on https://launch.joomla.org/ is set to daily renewal which means the site is lost in one day... monthly as default - like cloudaccess - works better I guess.

Thanks

**Comment by ghost** [comment]
@hardik-codes new Features goes only in J4

@martyharty please report the Request at CloudAccess.

**Comment by hardik-codes** [comment]
@franz-wohlkoenig so I have to make a PR for 4.0-dev ?

**Comment by ghost** [comment]
yes, there are only Bugfixes in J3.

**Comment by ghost** [comment]
closed as having Pull Request #24409

**Comment by Quy** [comment]
Fixed per #34634 for J4.
